cmd/verify/verify.go

```diff
@@ -0,0 +1,213 @@
+			}
+			return nil
+		},
+	}
+	clientConfig = cli.AddKubectlFlagsToCmd(command)
+	command.Flags().StringVar(&kubeconfigPath, "kubeconfigPath", "kubeconfig", "kubeconfig file location")
+	return command
+}
+
+// Check if kubectl is installed and the kubeconfig is pointing to the correct kubeconfig
+func verifyKubectl(kubeconfigPath string) (bool, error) {
+	_, err := exec.LookPath("kubectl")
+	if err != nil {
+		return false, ErrKubectlInstall
+	}
+
+	//Check if kubeconfig is correct and kubectl commands are functional
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return false, ErrKubectlSet
+	}
+	errClusterInfo := checkNamespace(config, "kube-system")
+	if errClusterInfo != nil {
+		return false, ErrClusterInfo
+	}
+	return true, nil
+}
+
+// Check if argocd cli is installed and the account has admin access
+func verifyArgoCD() (bool, error) {
+	// Check if argocd is installed
+	_, err := exec.LookPath("argocd")
+	if err != nil {
+		return false, ErrArgoCD
+	}
+
+	//Check if argocd has access and auth-token is not expired
+	_, errAcc := exec.Command("argocd", "account", "list").Output()
+	if errAcc != nil {
+		return false, ErrArgoCDAuthToken
+	}
+	return true, nil
+}
+
+// Check if git cli is installed
+func verifyGit() (bool, error) {
+	_, err := exec.LookPath("git")
+	if err != nil {
+		return false, ErrGit
+	}
+	return true, nil
+}
+
+// Verify if the arlon services are running
+func verifyArlon(config *restclient.Config) (bool, error) {
+	err := checkNamespace(config, "arlon")
+	if err == ErrListNs {
+		return false, ErrListNs
+	} else if err == ErrNs {
+		return false, ErrArlonNs
+	}
+	return true, nil
+}
+
+// Verify if capi services are running
+func verifyCapi(config *restclient.Config) (bool, error) {
+	errCapi := checkNamespace(config, "capi-system")
+	if errCapi != nil {
+		return false, ErrCapi
+	}
+
+	// Check for capa-system namespace
+	errCapaCP := checkCapaCloudProvider(config)
+	if errCapaCP == ErrNs {
+		//Check for capz-system incase capa-system is not present
+		errCapaZ := checkCapzCloudProvider(config)
+		if errCapaZ == ErrNs {
+			return false, ErrCapiCP
+		} else if errCapaZ == ErrListNs {
+			return false, ErrListNs
+		}
+	}
+
+	return true, nil
+
+}
+
+// Function to check capa-system namespace
+func checkCapaCloudProvider(config *restclient.Config) error {
+	errCapa := checkNamespace(config, "capa-system")
+	if errCapa != nil {
+		return errCapa
+	}
+	return nil
+}
+
+// Function to check capz-system namespace
+func checkCapzCloudProvider(config *restclient.Config) error {
+	errCapz := checkNamespace(config, "capz-system")
+	if errCapz != nil {
+		return errCapz
+	}
+	return nil
+}
+
+// Function to check for a particular namespace
+func checkNamespace(config *restclient.Config, namespace string) error {
+	kubeClient := kubernetes.NewForConfigOrDie(config)
+	corev1 := kubeClient.CoreV1()
+	nsList, err := corev1.Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
Use `Get()` with the namespace name, instead of Listing and iterating.

Reply:
Done